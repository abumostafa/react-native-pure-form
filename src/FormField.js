import React from 'react'
import PropTypes from 'prop-types'

class FormField extends React.Component {
  constructor(props) {
    super(props)
    this.state = {value: props.value || null, dirty: false, valid: false}
  }

  _onChange(value) {
    const {onChange} = this.props
    onChange && onChange(this.props.name, value)
    this.setState({dirty: true, value})
  }

  render() {
    const {component: WrappedComponent, ...props} = this.props
    return (
      <WrappedComponent
        {...props}
        onChange={value => this._onChange(value)}
        dirty={this.state.dirty}
        valid={this.props.valid}
        error={this.props.error}
        value={this.state.value}
      />
    )
  }
}

FormField.propTypes = {
  component: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.func,
  ]).isRequired,
  name: PropTypes.string.isRequired
}

export default FormField
