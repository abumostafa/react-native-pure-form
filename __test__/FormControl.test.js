import React from 'react'
import {TextInput} from 'react-native'
import renderer from 'react-test-renderer'
import FormField from '../src/FormField'

it('renders correctly', () => {
  const tree = renderer.create(<FormField component={TextInput} name="test-input" />).toJSON()
  expect(tree).toMatchSnapshot()
})
