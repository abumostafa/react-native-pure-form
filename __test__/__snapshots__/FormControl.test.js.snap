// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<TextInput
  allowFontScaling={true}
  dirty={false}
  name="test-input"
  onChange={[Function]}
  value={null}
/>
`;
